core/trino-main/src/main/java/io/trino/sql/planner/rowpattern/IrAnchorPattern.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner.rowpattern;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+import static io.trino.sql.planner.rowpattern.IrAnchorPattern.Type.PARTITION_START;
+import static java.util.Objects.requireNonNull;
+
+public class IrAnchorPattern
```

Comment:
move all the IR classes to an `ir` package under this one to separate them from the planner logic.